.gitlab-ci.yml

```diff
@@ -110,16 +110,52 @@ static-analysis:
   stage: analysis
   variables:
     DETEKT_PUBLIC_API: "true"
-    DETEKT_CUSTOM_RULES_BUILD_TASK: "assembleLibrariesRelease :tools:detekt:jar"
-    DETEKT_CUSTOM_RULES_JAR_PATH: "tools/detekt/build/libs/detekt.jar"
-    DETEKT_CUSTOM_RULES_YML_PATH: "detekt_custom.yml"
+    #    DETEKT_CUSTOM_RULES_BUILD_TASK: "assembleLibrariesRelease :tools:detekt:jar"
+    #    DETEKT_CUSTOM_RULES_JAR_PATH: "tools/detekt/build/libs/detekt.jar"
+    #    DETEKT_CUSTOM_RULES_YML_PATH: "detekt_custom.yml"
     DETEKT_GENERATE_CLASSPATH_BUILD_TASK: "printSdkDebugRuntimeClasspath"
     DETEKT_CLASSPATH_FILE_PATH: "sdk_classpath"
     FLAVORED_ANDROID_LINT: ":tools:lint:lint"
   trigger:
     include: "https://gitlab-templates.ddbuild.io/mobile/v34714656-060be019/static-analysis.yml"
     strategy: depend
 
+analysis:detekt-custom:
+  tags:
+    - "arch:amd64"
+  image: $CI_IMAGE_DOCKER
+  stage: analysis
+  timeout: 1h
+  script:
+    - ./gradlew assembleLibrariesRelease --stacktrace
+    - ./gradlew :tools:detekt:jar --stacktrace
+    - unzip dd-sdk-android-core/build/outputs/aar/dd-sdk-android-core-release.aar -d dd-sdk-android-core/build/outputs/aar/
+    - unzip features/dd-sdk-android-session-replay/build/outputs/aar/dd-sdk-android-session-replay-release.aar -d features/dd-sdk-android-session-replay/build/outputs/aar/
+    - ./gradlew :dd-sdk-android-core:printDetektClasspath --stacktrace
+    - curl -sSLO https://github.com/detekt/detekt/releases/download/v1.23.4/detekt-cli-1.23.4-all.jar
+    - java -jar detekt-cli-1.23.4-all.jar --config detekt_custom.yml --plugins "tools/detekt/build/libs/detekt.jar" -i "dd-sdk-android-core" -ex "**/*.kts" --jvm-target 11 -cp "$(cat dd-sdk-android-core/detekt_classpath)"
+    - java -jar detekt-cli-1.23.4-all.jar --config detekt_custom.yml --plugins "tools/detekt/build/libs/detekt.jar" -i "features/dd-sdk-android-logs" -ex "**/*.kts" --jvm-target 11 -cp "$(cat features/dd-sdk-android-logs/detekt_classpath):dd-sdk-android-core/build/outputs/aar/classes.jar"
+    - java -jar detekt-cli-1.23.4-all.jar --config detekt_custom.yml --plugins "tools/detekt/build/libs/detekt.jar" -i "features/dd-sdk-android-ndk" -ex "**/*.kts" --jvm-target 11 -cp "$(cat features/dd-sdk-android-ndk/detekt_classpath):dd-sdk-android-core/build/outputs/aar/classes.jar"
```

Comment:
maybe we should put this invocation in the gradle task? this will allow us to avoid a lot of repetitive lines and where only the name of the module is changed.